GDJS/Runtime/spriteruntimeobject.js

```diff
@@ -13,31 +69,51 @@
  * @memberof gdjs
  * @class SpriteAnimationFrame
  * @param {gdjs.ImageManager} imageManager The game image manager
- * @param {Object} frameData the data to be used to create the frame.
+ * @param {FrameData} frameData The frame data used to initialize the frame
  */
+
 gdjs.SpriteAnimationFrame = function(imageManager, frameData)
 {
+    /** @type {string} */
     this.image = frameData ? frameData.image : ""; //TODO: Rename in imageName, and do not store it in the object?
+    /** @type {PIXI.Texture} */
```

Comment:
This is not a PIXI Texture if the renderer is the one of Cocos2d :/ I don't have a good solution though.

Reply:
Last time someone checked it Cocos2D building didn't even work.

Reply:
Might be an issue, but let's try to stay generic. PIXI/Cocos aside, the class of the engine should be agnostic of the renderer - this means that the classes should not even know they are being rendered by something called PIXI. In the case they are storing something that comes from the renderer, it should treat this as an "opaque type": something that is unknown and just stored and passed back to the renderer that "know" what this thing is.

Reply:
Will add typing to renderer in another PR.